dataprofiler/profilers/profile_builder.py

```diff
@@ -51,56 +52,80 @@ def __init__(self, df_series, sample_size=None, min_sample_size=5000,
         :param options: Options for the structured profiler.
         :type options: StructuredOptions Object
         """
+        self.name = None
         self.options = options
         self._min_sample_size = min_sample_size
         self._sampling_ratio = sampling_ratio
         self._min_true_samples = min_true_samples
         if self._min_true_samples is None:
             self._min_true_samples = 0
 
-        # if you create your own DF without giving the column name,
-        # it labels the name as an int64, however, if you try to
-        # `json.dump` an int64, it errors.
-        if isinstance(df_series.name, str) or df_series.name is None:
-            self.name = df_series.name
-        else:
-            self.name = int(df_series.name)
-
         self.sample_size = 0
         self.sample = list()
         self.null_count = 0
         self.null_types = list()
         self.null_types_index = {}
-                
-        if not sample_size:
-            sample_size = self._get_sample_size(df_series)
-        if sample_size < len(df_series):
-            warnings.warn("The data will be profiled with a sample size of {}. "
-                          "All statistics will be based on this subsample and "
-                          "not the whole dataset.".format(sample_size))
-        clean_sampled_df, base_stats = \
-            self.get_base_props_and_clean_null_params(
-                df_series, sample_size, sample_ids=sample_ids)
-        self._update_base_stats(base_stats)
-
         self.profiles = {
-            'data_type_profile':
-            ColumnPrimitiveTypeProfileCompiler(
-                clean_sampled_df, self.options, pool),
-            'data_stats_profile':
-            ColumnStatsProfileCompiler(
-                clean_sampled_df, self.options, pool)
+            'data_type_profile': None,
+            'data_stats_profile': None
         }
 
-        use_data_labeler = True
-        if options and isinstance(options, StructuredOptions):
-            use_data_labeler = options.data_labeler.is_enabled
+        if df_series is not None and len(df_series) > 0:
+            
+            if not sample_size:
+                sample_size = self._get_sample_size(df_series)
+            if sample_size < len(df_series):
+                warnings.warn("The data will be profiled with a sample size of {}. "
+                              "All statistics will be based on this subsample and "
+                              "not the whole dataset.".format(sample_size))
+                
+            clean_sampled_df, base_stats  = \
+                self.clean_data_and_get_base_stats(
+                    df_series=df_series, sample_size=sample_size,
+                    min_true_samples=self._min_true_samples, sample_ids=sample_ids)
+            self.update_type_statistics(clean_sampled_df, pool)
+            self._update_base_stats(base_stats)
+
+    def update_type_statistics(self, clean_sampled_df, pool):
```

Comment:
nitpick: can we call this update profile compilers or update_compilers or something along those lines